src/client/ClientService.ts

```diff
@@ -87,48 +81,40 @@ class ClientService {
       },
       logger: logger.getChild(WebSocketServer.name),
     });
-
-    const clientService = new ClientService({
-      rpcServer,
-      webSocketServer,
-      logger,
-    });
-    await clientService.start({
-      options: {
-        host,
-        port,
-      },
-    });
-    logger.info(`Created ${this.name}`);
-    return clientService;
   }
+  public async start({
+                       manifest,
+                       options: {
+                         host = config.defaultsUser.clientServiceHost,
+                         port = config.defaultsUser.clientServicePort,
+                       },
+                     }: {
+    manifest: ServerManifest;
+    options: {
+      host?: string;
+      port?: number;
+    };
+  }): Promise<void> {
```

Comment:
Did you move the start method? It shouldn't be moved.

Reply:
Wdym by moved?

Reply:
The start method has been moved to just after the constructor? the order should be something like

1. properties
2. handlers
3. constructor
4. getters and setters
5.  async init functions like start and stop
6. public methods
7. private methods

Refer to how other start stop classes are structured.

Reply:
Did you fix this?

Reply:
Done.